third_party/move/move-prover/boogie-backend/src/spec_translator.rs

```diff
@@ -846,22 +846,20 @@ impl<'env> SpecTranslator<'env> {
             Operation::SpecFunction(module_id, fun_id, memory_labels) => {
                 self.translate_spec_fun_call(node_id, *module_id, *fun_id, args, memory_labels)
             },
-            Operation::Pack(_, _, Some(_)) => {
-                self.error(&loc, "variants not yet supported");
+            Operation::Pack(mid, sid, Some(variant)) => {
+                self.translate_pack_variant(node_id, *mid, *sid, variant, args)
             },
             Operation::Pack(mid, sid, None) => self.translate_pack(node_id, *mid, *sid, args),
             Operation::Tuple if args.len() == 1 => self.translate_exp(&args[0]),
             Operation::Tuple => self.error(&loc, "Tuple not yet supported"),
             Operation::Select(module_id, struct_id, field_id) => {
                 self.translate_select(node_id, *module_id, *struct_id, *field_id, args)
             },
-            Operation::SelectVariants(_module_id, _struct_id, _field_ids) => {
-                // TODO(#13806): implement for variants
-                self.error(&loc, "selection from variants no yet supported");
+            Operation::SelectVariants(module_id, struct_id, field_ids) => {
+                self.translate_select(node_id, *module_id, *struct_id, field_ids[0], args);
```

Comment:
@rahxephon89 Why is it okay to use `field_ids[0]` here (each field id is variant specific)?

Reply:
@rahxephon89  But if there are two variants `A` and `B`, and we select the field `q`, the field names are `A.q` and `B.q` (as opposed to just `q` if this were a struct's field selection). Maybe this is not an issue as you are going through the field env to get the name. Might be good to double check.